src/Umbraco.Cms.Integrations.DAM.Aprimo/Services/AprimoService.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System.Dynamic;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Text.Json;
+
+using Umbraco.Cms.Integrations.DAM.Aprimo.Models;
+
+namespace Umbraco.Cms.Integrations.DAM.Aprimo.Services
+{
+    public class AprimoService : IAprimoService
+    {
+        private readonly IHttpClientFactory _httpClientFactory;
+
+        private readonly OAuthConfigurationStorage _oauthConfigurationStorage;
+
+        public AprimoService(IHttpClientFactory httpClientFactory, OAuthConfigurationStorage oauthConfigurationStorage)
+        {
+            _httpClientFactory = httpClientFactory;
+
+            _oauthConfigurationStorage = oauthConfigurationStorage;
+        }
+
+        public async Task<AprimoResponse<Record>> SearchRecord(Guid id)
+        {
+            var client = _httpClientFactory.CreateClient(Constants.AprimoClient);
+
+            var oauthConfiguration = _oauthConfigurationStorage.Get();
+            if (oauthConfiguration == null) return AprimoResponse<Record>.Fail(Constants.ErrorResources.Unauthorized, false);
+
+            client.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue("Umbraco.Cms.Integrations.DAM.Aprimo", "1.0.0"));
```

Comment:
Looks like some of this code is repeated in the other `SearchRecords` method - so seems like you could create some private helper methods to DRY this up.

Reply:
Updated